code/modules/mob/living/carbon/human/human_defense.dm

```diff
@@ -140,7 +140,19 @@
 		throwpower = I.throwforce
 		if(I.thrownby == src) //No throwing stuff at yourself to trigger hit reactions
 			return ..()
-	if(check_shields(throwpower, "\the [AM.name]", AM, THROWN_PROJECTILE_ATTACK))
+	if(martial_art && martial_art.catch_chance)
+		if(prob(martial_art.catch_chance))
+			if(canmove && !restrained() && I && isturf(I.loc))
+				var/thrower = I.thrownby
+				if(put_in_hands(I))
+					visible_message("<span class='warning'>[src] catches [I]!</span>")
+					if(!(a_intent == "help") && thrower && thrower in view(src))
+						sleep(1)

```

Comment:
why are you sleeping here, this will mean that for at least one tick the item will be both in the thrower and catchers hand, which could lead to it remaining in two places at once, or other absurd logical issues (i.e if they place it back into their backpack fast enough)

you should completely reorder this to ensure it is removed from the throwers hands first and then placed into the catchers hands. With no sleep between those two events occuring